drivers/sensor/ism330dhcx/ism330dhcx.c

```diff
@@ -0,0 +1,844 @@
+
+	idx = ism330dhcx_shub_get_idx(SENSOR_CHAN_PRESS);
+	if (idx < 0) {
+		LOG_DBG("external press/temp not supported");
+		return;
+	}
+
+	raw_val = sys_le32_to_cpu((s32_t)(data->ext_data[idx][0] |
+					  (data->ext_data[idx][1] << 8) |
+					  (data->ext_data[idx][2] << 16)));
+
+	/* Pressure sensitivity is 4096 LSB/hPa */
+	/* Convert raw_val to val in kPa */
+	val->val1 = (raw_val >> 12) / 10;
+	val->val2 = (raw_val >> 12) % 10 * 100000 +
+		(((s32_t)((raw_val) & 0x0FFF) * 100000L) >> 12);
+}
+
+static inline void ism330dhcx_temp_convert(struct sensor_value *val,
+					   struct ism330dhcx_data *data)
+{
+	s16_t raw_val;
+	int idx;
+
+	idx = ism330dhcx_shub_get_idx(SENSOR_CHAN_PRESS);
+	if (idx < 0) {
+		LOG_DBG("external press/temp not supported");
+		return;
+	}
+
+	raw_val = sys_le16_to_cpu((s16_t)(data->ext_data[idx][3] |
+					  (data->ext_data[idx][4] << 8)));
+
+	/* Temperature sensitivity is 100 LSB/deg C */
+	val->val1 = raw_val / 100;
+	val->val2 = (s32_t)raw_val % 100 * (10000);
+}
+#endif
+
+static int ism330dhcx_channel_get(struct device *dev,
+				  enum sensor_channel chan,
+				  struct sensor_value *val)
+{
+	struct ism330dhcx_data *data = dev->driver_data;
+
+	switch (chan) {
+	case SENSOR_CHAN_ACCEL_X:
+	case SENSOR_CHAN_ACCEL_Y:
+	case SENSOR_CHAN_ACCEL_Z:
+	case SENSOR_CHAN_ACCEL_XYZ:
+		ism330dhcx_accel_channel_get(chan, val, data);
+		break;
+	case SENSOR_CHAN_GYRO_X:
+	case SENSOR_CHAN_GYRO_Y:
+	case SENSOR_CHAN_GYRO_Z:
+	case SENSOR_CHAN_GYRO_XYZ:
+		ism330dhcx_gyro_channel_get(chan, val, data);
+		break;
+#if defined(CONFIG_ISM330DHCX_ENABLE_TEMP)
+	case SENSOR_CHAN_DIE_TEMP:
+		ism330dhcx_gyro_channel_get_temp(val, data);
+		break;
+#endif
+#if defined(CONFIG_ISM330DHCX_SENSORHUB)
+	case SENSOR_CHAN_MAGN_X:
+	case SENSOR_CHAN_MAGN_Y:
+	case SENSOR_CHAN_MAGN_Z:
+	case SENSOR_CHAN_MAGN_XYZ:
+		ism330dhcx_magn_get_channel(chan, val, data);
+		break;
+
+	case SENSOR_CHAN_HUMIDITY:
+		ism330dhcx_hum_convert(val, data);
+		break;
+
+	case SENSOR_CHAN_PRESS:
+		ism330dhcx_press_convert(val, data);
+		break;
+
+	case SENSOR_CHAN_AMBIENT_TEMP:
+		ism330dhcx_temp_convert(val, data);
+		break;
+#endif
+	default:
+		return -ENOTSUP;
+	}
+
+	return 0;
+}
+
+static const struct sensor_driver_api ism330dhcx_api_funcs = {
+	.attr_set = ism330dhcx_attr_set,
+#if CONFIG_ISM330DHCX_TRIGGER
+	.trigger_set = ism330dhcx_trigger_set,
+#endif
+	.sample_fetch = ism330dhcx_sample_fetch,
+	.channel_get = ism330dhcx_channel_get,
+};
+
+static int ism330dhcx_init_chip(struct device *dev)
+{
+	struct ism330dhcx_data *ism330dhcx = dev->driver_data;
+	u8_t chip_id;
+
+	if (ism330dhcx_device_id_get(ism330dhcx->ctx, &chip_id) < 0) {
+		LOG_DBG("Failed reading chip id");
+		return -EIO;
+	}
+
+	LOG_INF("chip id 0x%x", chip_id);
+
+	if (chip_id != ISM330DHCX_ID) {
+		LOG_DBG("Invalid chip id 0x%x", chip_id);
+		return -EIO;
+	}
+
+	/* reset device */
+	if (ism330dhcx_reset_set(ism330dhcx->ctx, 1) < 0) {
+		return -EIO;
+	}
+
+	k_busy_wait(100);
```

Comment:
is there a reason you didn't use `ism330dhcx_reboot` here?

Reply:
They are two different functions:
1. reboot is loading the memory content of the sensor
2. reset the registers to default. Generally is enough.